packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -0,0 +1,114 @@
+
+            // The interface for the new URL search params
+            const newParams = new URLSearchParams();
+
+            /**
+             * Function that handles the presence of query parameters
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParams(queryParam) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                }
+            }
+
+            /**
+             * Function that handles the presence of query parameters and props
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParamsAndProps(queryParam, prop) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                } else if (prop) {
+                    if (prop === primaryColorProp) {
+                        newParams.append(queryParam, primaryColorProp.replace("#", ""))
+                    } else {
+                        newParams.append(queryParam, prop)
+                    }
+                }
+            }
+
+            // Handle query parameters for the gmApiKey and mapboxAccessToken
+            handleQueryParams('gmApiKey');
+            handleQueryParams('mapboxAccessToken');
+
+            // Handle query parameters and props for apiKey, primaryColor, logo 
+            handleQueryParamsAndProps('apiKey', apiKeyProp);
+            handleQueryParamsAndProps('primaryColor', primaryColorProp);
+            handleQueryParamsAndProps('logo', logoProp);
+
+            // Get the string with all the final query parameters
+            const finalParams = newParams.toString()
+
+            // Construct the QR code URL
+            let QRCodeURL = `${targetUrl}/?${finalParams}&directionsFrom=${directionsFrom.id}&directionsTo=${directionsTo.id}`
+
+            const options = {
+                errorCorrectionLevel: 'L',
+                margin: 0,
+                width: '225'
+            };
+
+            QRCode.toDataURL(QRCodeURL, options)
+                .then((dataUrl) => {
+                    elementRef.current.src = dataUrl;
+                }).catch(() => {
+                    console.log('error')
+                });
+        }
+
+    }, [directionsFrom, directionsTo])
+
+    return (<>
+        <div className="background"></div>
+        <div className="qr-code">
+            <img alt="QR Code" className="qr-code__image" ref={elementRef} />
+            <p>Scan the QR code to see the route on your phone</p>
```

Comment:
We will probably need to get this in French, German and Danish. Let's see what goes into main first 😆

Reply:
💨🏃🏻 Run!

Reply:
I guess you won 😭

Reply:
Hahah 😂 Oops